napari/components/scale_bar.py

```diff
@@ -28,14 +36,31 @@ class ScaleBar(EventedModel):
         then it has the color opposite of this color.
     font_size : float
         The font size (in points) of the text.
+    box : bool
+        If background box is visible or not.
+    box_color : Optional[str | array-like]
+        Box color. Can be any color name recognized by vispy or
+        hex value if starting with `#`. If array-like must be 1-dimensional
+        array with 3 or 4 elements.
     unit : Optional[str]
         Unit to be used by the scale bar. The value can be set
         to `None` to display no units.
     """
 
     visible: bool = False
     colored: bool = False
+    color: Optional[Array[float, (4,)]] = None
     ticks: bool = True
     position: Position = Position.BOTTOM_RIGHT
     font_size: float = 10
+    box: bool = False
+    box_color: Optional[Array[float, (4,)]] = None
     unit: Optional[str] = None
+
+    @validator('color', pre=True)
```

Comment:
The way this validator is defined makes passing a parameter to the initializer inconsistent with setting the field. I.e.

```python
sb = ScaleBar(color=None)
sb.color # => None

sb.color = None
sb.color # => array([0, 0, 0, 0])
```

That's because `transform_color(None)` returns `array([[0, 0, 0, 0]])` rather than `None`.

It would probably be a good idea to make these consistent. I can think of a couple of options.

1. Make the color field `Array[float, (4,)]` and assign assign the default color (from the vispy code) as the default value. As you have the `colored` field to indicate whether this value should be used or not, using an `Optional` to indicate some default color gets a little confusing.
2. Change the validator function to return `None` if `v is None`. But then you'll need to call `transform_color(color)[0]` when `color is None`.

I have a mild preference for (1) because it simplifies the logic in a few places for me, but I don't feel strongly.

There is a similar issue for `box_color`.